src/Data/Lens/Prism.purs

```diff
@@ -1,7 +1,46 @@
--- | This module defines functions for working with prisms.
+-- | Prisms are used for selecting cases of a type, most often a sum
```

Comment:
Law abiding prisms only apply to types which are _isomorphic_ to sums. I wonder if that could be brought out here. Maybe it's not that important though.

Reply:
Generally, I'd prefer to have all laws explained in one file, because the law differences between different optics are instructive.

In this particular case, I envision that the [sum-type-specific examples file](https://github.com/purescript-contrib/purescript-profunctor-lenses/blob/master/examples/src/PrismsForSumTypes.purs) will be joined with a file about using prisms on things isomorphic to sum types.